### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by rickardl**

### CODE REVIEW

File: examples/default/example.tf [hunk 1]
```diff
@@ -1,5 +1,36 @@
 provider "aws" {
-  region = "eu-west-1"
+  version = "1.20.0"
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: Maybe bump to `1.30.0` which is the latest version?

File: examples/default/example.tf [hunk 2]
```diff
@@ -10,5 +41,122 @@ data "aws_subnet_ids" "main" {
   vpc_id = "${data.aws_vpc.main.id}"
 }
 
-# REST OF THE EXAMPLE
+module "alb" {
+  source  = "telia-oss/loadbalancer/aws"
+  version = "0.1.0"
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: Bump to `0.1.1`

File: examples/default/example.tf [hunk 3]
```diff
@@ -10,5 +41,122 @@ data "aws_subnet_ids" "main" {
...
+
+  tags {
+    environment = "test"
+    terraform   = "true"
+  }
+}
+
+module "rds-instance" {
+  source  = "telia-oss/rds-instance/aws"
+  version = "0.1.1"
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: Bump to `0.2.0`

File: examples/default/example.tf [hunk 4]
```diff
@@ -10,5 +41,122 @@ data "aws_subnet_ids" "main" {
...
+  vpc_id      = "${data.aws_vpc.main.id}"
+  subnet_ids  = "${data.aws_subnet_ids.main.ids}"
+
+  tags {
+    environment = "test"
+    terraform   = "true"
+  }
+}
+
+module "rds-instance" {
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: We try to always use `_` instead of `-` for naming resources.

File: examples/default/example.tf [hunk 5]
```diff
@@ -10,5 +41,122 @@ data "aws_subnet_ids" "main" {
...
+
+  task_container_environment = {
+    "GF_SERVER_ROOT_URL"      = "${local.server_protocol}://${module.alb.dns_name}:${local.public_port}"
+    "GF_SERVER_PROTOCOL"      = "${local.server_protocol}"
+    "GRAFANA_PLUGINS_ENABLED" = "${local.grafana_plugins_enabled}"
+
+    "GF_DATABASE_TYPE"     = "${local.rds_instance_engine}"
+    "GF_DATABASE_USER"     = "${local.rds_instance_username}"
+    "GF_DATABASE_PASSWORD" = "${local.rds_instance_password}"
+    "GF_DATABASE_HOST"     = "${module.rds-instance.endpoint}"
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: This needs to be changed if we rename `rds-instance` to `rds_instance`

File: examples/default/example.tf [hunk 6]
```diff
@@ -10,5 +41,122 @@ data "aws_subnet_ids" "main" {
...
+  }
+}
+
+# ----------------------------------------
+# Security Group Rules
+# ----------------------------------------
+
+resource "aws_security_group_rule" "lb_ingress" {
+  type              = "ingress"
+  description       = "HTTP"
```
Comment (mikael-lindstrom) [inline examples/default/example.tf]: Maybe change to `Allow HTTP from internet`?

File: main.tf [hunk 7]
```diff
@@ -2,3 +2,46 @@
...
+}
+
+# ----------------------------------------
+# Grafana ECS Fargate Service
+# ----------------------------------------
+
+module "grafana" {
+  source = "github.com/telia-oss/terraform-aws-ecs-fargate"
+
+  //source  = "telia-oss/ecs-fargate/aws"
```
Comment (mikael-lindstrom) [inline main.tf]: We should use this instead of the git url.

File: outputs.tf [hunk 8]
```diff
@@ -2,3 +2,21 @@
 # Output
 # ------------------------------------------------------------------------------
 
+output "grafana_URL" {
+  value = "https://fix.me"
```
Comment (mikael-lindstrom) [inline outputs.tf]: We should probably remove this since the module doesn't setup a Route53 domain.

File: variables.tf [hunk 9]
```diff
@@ -1,12 +1,81 @@
...
+  type        = "list"
+}
+
+variable "alb_arn" {
+  description = "Arn for the ALB for which the service should be attach to."
+}
+
+variable "task_container_image" {
+  description = "Image for the task definition (repo:tag or repo@digest)."
+  default     = "grafana/grafana"
```
Comment (mikael-lindstrom) [inline variables.tf]: Maybe we should default to at tag like `latest` for clarity?

File: variables.tf [hunk 10]
```diff
@@ -1,12 +1,81 @@
...
+  default     = 3000
+}
+
+variable "task_container_protocol" {
+  description = "Protocol that the container exposes."
+  default     = "HTTP"
+}
+
+variable "desired_count" {
+  description = "The dsfsfdf."
```
Comment (mikael-lindstrom) [inline variables.tf]: `The dsfsfdf`? :)

**Review by mikael-lindstrom — CHANGES_REQUESTED** [review]
Suggested some small changes but in general it look good 👍

File: examples/default/example.tf (hunk 1 above)
Comment (rickardl) [reply examples/default/example.tf]: 1.30.0 (August 02, 2018) Will update

File: variables.tf (hunk 10 above)
Comment (rickardl) [reply variables.tf]: 💃

**Review by mikael-lindstrom — APPROVED** [review]
LGTM 👍

**Pull request merged**